### PR TITLE
general: Elasticsearch logging with Lumberjack.

### PR DESCRIPTION
**Pull request by jmacmahon**
- Added disabled-by-default config options for Elasticsearch logging.
- When Elasticsearch logging is configured, don't populate rnkPAGEVIEWS and
  rnkDOWNLOADS.

Signed-off-by: Joe MacMahon joe.macmahon@cern.ch

### CODE REVIEW

**Comment by egabancho** [comment]
inveniosoftware/invenio/pull/2431

File: requirements.txt [hunk 1]
```diff
@@ -28,3 +28,4 @@ jinja2==2.7.2
 redis==2.9.0
 nydus==0.10.6
 Cerberus==0.5
+lumberjack==git+https://github.com/jmacmahon/lumberjack.git
```
Comment (egabancho) [inline requirements.txt:31]: Could you update this one to point to CDS' repo?

**Comment by jmacmahon** [comment]
Done, but my branch isn't stable at the moment, so don't merge it all.

File: modules/bibdocfile/lib/bibdocfile.py [hunk 2]
```diff
@@ -2854,12 +2874,23 @@ def register_download(self, ip_address, version, docformat, userid=0, recid=0):
...
+                'id_bibrec': recid,
+                'id_bibdoc': self.id,
+                'file_version': version,
+                'file_format': docformat,
+                'id_user': userid,
+                'client_host': ip_address
+            }
+            _DOWNLOAD_LOG.info(log_entry)
+        else:
+            return run_sql("INSERT DELAYED INTO rnkDOWNLOADS "
```
Comment (egabancho) [inline modules/bibdocfile/lib/bibdocfile.py:2888]: Could you please remove the `DELAYED`? see 386ac9d5ef1041f1ac98725e0c3aea178d811f96

File: modules/bibrank/lib/bibrank_downloads_similarity.py [hunk 3]
```diff
@@ -55,10 +71,18 @@ def register_page_view_event(recid, uid, client_ip_address):
...
-                   (recid, uid, client_ip_address))
+    if CFG_ELASTICSEARCH_LOGGING:
+        log_event = {
+            'id_bibrec': recid,
+            'id_user': uid,
+            'client_host': client_ip_address
+        }
+        _PAGEVIEW_LOG.info(log_event)
+    else:
+        return run_sql("INSERT DELAYED INTO rnkPAGEVIEWS " \
```
Comment (egabancho) [inline modules/bibrank/lib/bibrank_downloads_similarity.py:82]: Same as before ;-)

**Comment by jmacmahon** [comment]
I've removed the INSERT DELAYED statements, so now the elasticsearch_logging branch should be as on production.

**Pull request merged**